File: algorithms/code/leetcode/lc026_remove_duplicates_from_sorted_array/LC026RemoveDuplicatesFromSortedArray.java
package leetcode.lc026_remove_duplicates_from_sorted_array;

/**
 * https://leetcode.com/problems/remove-duplicates-from-sorted-array/
 * #easy
 */
public final class LC026RemoveDuplicatesFromSortedArray {
    public int removeDuplicates(final int[] nums) {
        int counter = 0;
        for (int i = 1; i < nums.length; i++) {
            int spot = counter;
            if (nums[i] == nums[i - 1 - spot]) {
                counter++;
            }
            nums[i - spot] = nums[i];
        }
        return nums.length - counter;
    }
}

File: algorithms/code/leetcode/lc026_remove_duplicates_from_sorted_array/lc026_remove_duplicates_from_sorted_array.go
// Package lc026 implements https://leetcode.com/problems/remove-duplicates-from-sorted-array/
package lc026

// RemoveDuplicates function removes duplicates from a sorted array in place
func RemoveDuplicates(nums []int) int {
	if len(nums) == 0 {
		return 0
	}
	counter := 0
	for i := 1; i < len(nums); i++ {
		spot := counter
		if nums[i] == nums[i-1-spot] {
			counter++
		}
		nums[i-spot] = nums[i]
	}
	return len(nums) - counter
}
